server/mock/app.go

```diff
@@ -34,7 +34,7 @@ func testTxHandler(options middleware.TxHandlerOptions) tx.Handler {
 // similar to a real app. Make sure rootDir is empty before running the test,
 // in order to guarantee consistent results
 func NewApp(rootDir string, logger log.Logger) (abci.Application, error) {
-	db, err := sdk.NewLevelDB("mock", filepath.Join(rootDir, "data"))
+	db, err := sdk.NewDB("mock", "", filepath.Join(rootDir, "data"))
```

Comment:
This was the trickiest spot to update. It's the only spot where there wasn't an already-communicated db-backend setting. I dug through uses of `mock.NewApp` and couldn't find an easy and clear way let it be provided without adding complexity that probably isn't desired here.

So I opted to just use the default db backend (which is `goleveldb`). I figure that's going to be most uses since it's the default for `types.DBBackend`. It's also the safest since that one seems to be always available. The only time someone will see a difference is if they previously compiled and set `types.DBBackend` to something else, and then used `mock.NewApp` in that compiled code. I also figured that, since it's for mocking, always using the default isn't too bad.

Reply:
should this be memes to avoid using more resources than we should in testing

Reply:
Good call. I've changed it to `memdb`.

Reply:
amazing, love that you got my autocorrect 😆